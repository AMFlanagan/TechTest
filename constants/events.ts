import {
  IEvent,
  Sport,
  FootballMarket,
  RugbyMarket,
  HorseRacingMarket,
} from '../types'

export const allEvents: { [key in Sport]: IEvent[] } = {
  [Sport.FOOTBALL]: [
    {
      id: 1,
      sport: Sport.FOOTBALL,
      eventName: 'Leeds Vs Liverpool',
      startTime: '2022-01-02T15:00:00.000Z',
      markets: [
        {
          id: 1,
          type: FootballMarket.RESULT,
          outcomes: [
            {
              id: 100,
              name: 'home',
              odds: 6.333333333,
            },
            {
              id: 101,
              name: 'draw',
              odds: 4.0,
            },
            {
              id: 102,
              name: 'away',
              odds: 1.333333333,
            },
          ],
        },
      ],
    },
    {
      id: 2,
      sport: Sport.FOOTBALL,
      eventName: 'Chelsea Vs Manchester United',
      startTime: '2022-01-02T15:00:00.000Z',
      markets: [
        {
          id: 1,
          type: FootballMarket.RESULT,
          outcomes: [
            {
              id: 103,
              name: 'home',
              odds: 2.0,
            },
            {
              id: 104,
              name: 'draw',
              odds: 4.0,
            },
            {
              id: 105,
              name: 'away',
              odds: 3.0,
            },
          ],
        },
      ],
    },
    {
      id: 3,
      sport: Sport.FOOTBALL,
      eventName: 'Chester Vs Wrexham',
      startTime: '2022-01-02T15:00:00.000Z',
      markets: [
        {
          id: 1,
          type: FootballMarket.RESULT,
          outcomes: [
            {
              id: 106,
              name: 'home',
              odds: 1.333333333,
            },
            {
              id: 107,
              name: 'draw',
              odds: 3.0,
            },
            {
              id: 108,
              name: 'away',
              odds: 4.222222,
            },
          ],
        },
      ],
    },
  ],
  [Sport.RUGBY]: [
    {
      id: 4,
      sport: Sport.RUGBY,
      eventName: 'England Vs Wales',
      startTime: '2022-01-02T15:00:00.000Z',
      markets: [
        {
          id: 1,
          type: RugbyMarket.RESULT,
          outcomes: [
            {
              id: 109,
              name: 'home',
              odds: 6.333333333,
            },
            {
              id: 110,
              name: 'draw',
              odds: 4.0,
            },
            {
              id: 111,
              name: 'away',
              odds: 1.333333333,
            },
          ],
        },
      ],
    },
    {
      id: 5,
      sport: Sport.RUGBY,
      eventName: 'Scotland Vs France',
      startTime: '2022-01-02T15:00:00.000Z',
      markets: [
        {
          id: 1,
          type: RugbyMarket.RESULT,
          outcomes: [
            {
              id: 112,
              name: 'home',
              odds: 2.0,
            },
            {
              id: 113,
              name: 'draw',
              odds: 4.0,
            },
            {
              id: 114,
              name: 'away',
              odds: 3.0,
            },
          ],
        },
      ],
    },
  ],
  [Sport.HORSE_RACING]: [
    {
      id: 6,
      sport: Sport.HORSE_RACING,
      eventName: '1400 York',
      startTime: '2022-01-02T14:00:00.000Z',
      markets: [
        {
          id: 1,
          type: HorseRacingMarket.RESULT,
          outcomes: [
            {
              id: 115,
              name: 'Jerry Springy',
              odds: 6.0,
            },
            {
              id: 116,
              name: 'Chosen Mate',
              odds: 4.0,
            },
            {
              id: 117,
              name: 'Little Nelson',
              odds: 1.0,
            },
            {
              id: 118,
              name: 'Rock the world',
              odds: 1.2,
            },
            {
              id: 119,
              name: 'Red Stripe',
              odds: 11.0,
            },
          ],
        },
      ],
    },
  ],
}
