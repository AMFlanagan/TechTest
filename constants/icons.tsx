export enum IconName {
  UP = 'UP',
  CROSS = 'CROSS',
}

const Icons = {
  [IconName.UP]: (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      id="Outline"
      viewBox="0 0 24 24"
      width="512"
      height="512"
    >
      <path d="M23.71,16.29,15.54,8.12a5,5,0,0,0-7.08,0L.29,16.29a1,1,0,0,0,1.42,1.42L9.88,9.54a3,3,0,0,1,4.24,0l8.17,8.17a1,1,0,0,0,1.42,0A1,1,0,0,0,23.71,16.29Z" />
    </svg>
  ),
  [IconName.CROSS]: (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      id="Outline"
      viewBox="0 0 24 24"
      width="512"
      height="512"
    >
      <path d="M23.707.293h0a1,1,0,0,0-1.414,0L12,10.586,1.707.293a1,1,0,0,0-1.414,0h0a1,1,0,0,0,0,1.414L10.586,12,.293,22.293a1,1,0,0,0,0,1.414h0a1,1,0,0,0,1.414,0L12,13.414,22.293,23.707a1,1,0,0,0,1.414,0h0a1,1,0,0,0,0-1.414L13.414,12,23.707,1.707A1,1,0,0,0,23.707.293Z" />
    </svg>
  ),
}

export default Icons
