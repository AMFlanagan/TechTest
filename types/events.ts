export enum Sport {
  FOOTBALL = 'football',
  RUGBY = 'rugby',
  HORSE_RACING = 'horse-racing',
}

export enum FootballMarket {
  RESULT = 'Result',
  NEXT_GOAL = 'Next goal',
}

export enum RugbyMarket {
  RESULT = 'Result',
  NEXT_TRY = 'Next Try',
}

export enum HorseRacingMarket {
  RESULT = 'Result',
}

export interface IOutcome {
  id: number
  name: string
  odds: number
}

export type Market = FootballMarket | HorseRacingMarket | RugbyMarket

export interface IMarket {
  type: Market
  outcomes: IOutcome[]
}

export interface IEvent {
  id: number
  sport: Sport
  eventName: string
  markets: IMarket[]
  startTime: string
}
