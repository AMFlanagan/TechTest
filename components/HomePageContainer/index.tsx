import React from 'react'
import { IEvent, Sport } from '../../types'

import EventList from '../common/EventList'

interface IProps {
  allEvents: { [key in Sport]: IEvent[] }
}

const HomePage: React.FC<IProps> = (props: IProps) => {
  return (
    <div>
      {Object.entries(props.allEvents).map((sport, i: number) => {
        const [title, events] = sport
        return <EventList key={i} title={title} events={events} />
      })}
    </div>
  )
}

HomePage.displayName = 'HomePage'

export default HomePage
