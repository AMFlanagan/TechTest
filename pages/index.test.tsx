import { render } from '@testing-library/react'
import { allEvents } from '../constants/events'
import HomePage from '.'
import HomePageContainer from '../components/HomePageContainer'
import { NextPageContext } from 'next'

jest.mock('../components/HomePageContainer', () =>
  jest.fn(() => <div data-testid="HomePageContainer"></div>)
)

// @ts-ignore - TODO: Add global definitions
global.fetch = jest.fn(() =>
  Promise.resolve({
    json: () => Promise.resolve(allEvents),
  })
)

describe('HomePage', () => {
  it('renders as expect', () => {
    const { getByTestId, asFragment } = render(
      <HomePage allEvents={allEvents} />
    )

    expect(getByTestId('HomePageContainer')).toBeTruthy()
    expect(HomePageContainer).toBeCalledWith(
      {
        allEvents,
      },
      expect.anything()
    )
    expect(asFragment()).toMatchSnapshot()
  })

  describe('getInitialProps', () => {
    test('returns the correct props for the page', async () => {
      let props

      if (HomePage.getInitialProps) {
        props = await HomePage.getInitialProps({} as NextPageContext)
      }

      expect(props).toEqual({
        allEvents,
      })
    })
  })
})
