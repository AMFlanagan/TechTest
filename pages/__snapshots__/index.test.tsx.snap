// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage renders as expect 1`] = `
<DocumentFragment>
  <div
    data-testid="HomePageContainer"
  />
</DocumentFragment>
`;
